packages/lib-react-components/src/AnimatedNumber/AnimatedNumber.spec.js

```diff
@@ -0,0 +1,187 @@
+  let oldIntersectionObserver
+  /*
+  * The tests run in Node, so we need to mock the IntersectionObserver
+  * and manually set mock entries.
+  */
+  const mockEntries = [{
+    isIntersecting: true,
+    intersectionRatio: 1
+  }]
+
+  class MockIntersectionObserver {
+    constructor(fn) {
+      this.fn = fn
+    }
+    observe() {
+      this.fn(mockEntries)
+    }
+    unobserve() { }
+    disconnect() { }
+  }
+
+  beforeEach(function () {
+    oldIntersectionObserver = window.IntersectionObserver
+    window.IntersectionObserver = MockIntersectionObserver
+  })
+
+  afterEach(function () {
+    window.IntersectionObserver = oldIntersectionObserver
+  })
+
+  describe('without any animation', function () {
+    describe('default behaviour', function () {
+      it('should animate the number', async function () {
+        const Story = composeStory(Stories.Default, Meta)
+        render(<Story />)
+        expect(prefersReducedMotion()).to.be.true()
+        const number = document.querySelector('span')
+        expect(number.textContent).to.equal('0')
+        await waitFor(() => {
+          expect(number.textContent).to.equal('123,456')
+        })
+      })
+    })
+    
+    describe('with 0', function () {
+      it('should not animate the number', async function () {
+        const Story = composeStory(Stories.Zero, Meta)
+        render(<Story />)
+        expect(prefersReducedMotion()).to.be.true()
+        const number = document.querySelector('span')
+        expect(number.textContent).to.equal('0')
+        await waitFor(() => {
+          expect(number.textContent).to.equal('0')
+        })
+      })
+    })
+
+    describe('updating after the animation', function () {
+      it('should update the number', async function () {
+        const user = userEvent.setup()
+        const Story = composeStory(Stories.UpdateTheValue, Meta)
+        render(<Story />)
+        expect(prefersReducedMotion()).to.be.true()
+        const number = document.querySelector('span')
+        expect(number.textContent).to.equal('0')
+        await waitFor(() => {
+          expect(number.textContent).to.equal('10')
+        })
+        await user.click(document.querySelector('button'))
+        await waitFor(() => {
+          expect(number.textContent).to.equal('11')
+        })
+      })
+    })
+  
+    describe('with a deferred value', function () {
+      it('should animate the number', async function () {
+        const clock = sinon.useFakeTimers({
+          shouldClearNativeTimers: true
+        })
+        const Story = composeStory(Stories.DeferredAnimation, Meta)
+        render(<Story />)
+        expect(prefersReducedMotion()).to.be.true()
+        const number = document.querySelector('span')
+        expect(number.textContent).to.equal('0')
+        // wait 2000ms for the value prop to change
+        clock.tick(2000)
+        clock.restore()
+        expect(number.textContent).to.equal('0')
```

Comment:
This is on me, but to confirm my understanding it's still `'0'` after 2 seconds because it's animating, then it's the new value (line 100) after the animation has concluded, even though the new value was passed to the component at 2 seconds (per DeferredAnimation story).

Reply:
It’s 0 here because it’s still waiting to start the animation. The `value` changes at 2s, which starts the animation. Previously, the animation started while the value was 0, and d3 would insert a 0 into the element after a 1s wait, overwriting the actual value. 

In the storybook, you can see that by scrolling the animated number into view before `value` changes. Previously, the displayed value stayed at 0, but now it starts animating once the prop changes.